src/main/java/com/google/cloud/resourcemanager/Lien.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.resourcemanager;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.api.gax.paging.Page;
+import com.google.cloud.resourcemanager.ResourceManager.LienListOption;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * A Google Cloud Resource Manager lien object.
+ *
+ * <p>A Lien represents an encumbrance on the actions that can be performed on a resource. This
+ * class' member variables are immutable. Methods that change or update the underlying Lien
+ * information return a new Lien instance. {@code Lien} adds a layer of service-related
```

Comment:
" This class' member variables are immutable." => " This is an immutable class."

Reply:
done